macleann/enchiridion-server
# Adding likes functionality

# Implementing 'Likes' Feature for Playlists

This has been on my list for some time now, so I'm glad I finally got around to it. I had to run some migrations after making some new models. I had the idea to include trending playlists, which will sort by the number of likes and date range. I went ahead and included trending section on the homepage on the client.

### Bug fixes:

Fixes #16 Feature: Allow logged in user to like another user's playlist
Fixes #17 Feature: Allow any user to see all playlists' total count of likes

## Type of change


# How to Test?

Make sure you've followed all installation and setup instructions in the README.

```
git fetch origin nm-likes
git checkout nm-likes
python manage.py runserver
```

- [ ] Send a `POST` request to `localhost:8000/register' with the body:
{
     "username": "your_username",
     "email": "your_email",
     "password": "your_password",
     "firstName": "your_first_name",
     "lastName": "your_last_name",
}
- [ ] Postman should store some cookies for you that you can use for the following request. No need for setting authorization headers!
- [ ] Next, send a `POST` request to `http://localhost:8000/likes`. In the body, put the id of an existing playlist:
{
    "playlist_id": 1
}
- [ ] You should receive a response that says { "message": "Liked" }.
- [ ] Then you can unlike that playlist by sending a delete request to `http://localhost:8000/likes/1` <-- make sure the id matches the one you just liked.
- [ ] You should get a response that says { "message": "Unliked" }.
- [ ] If you try to unlike a playlist that you've not liked previously, I've added error handling for that. I'm realizing now that I don't have any error handling for liking a playlist that doesn't exist. Next PR I suppose.


# Checklist:
